lib/puppet/provider/service/systemd.rb

```diff
@@ -7,6 +7,7 @@
 
   defaultfor :osfamily => [:archlinux]
   defaultfor :osfamily => :redhat, :operatingsystemmajrelease => "7"
+  defaultfor :osfamily => :redhat, :operatingsystem => :fedora, :operatingsystemmajrelease => ["17", "18", "19", "20", "21"]
```

Comment:
Other than the comment I just added, no.  I can live with having explicit version number checks for now.